dotm_search.py

```diff
@@ -1,15 +1,50 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
+import os
+import argparse
+import zipfile
+parser = argparse.ArgumentParser()
+
 
 """
 Given a directory path, search all files in the path for a given text string
 within the 'word/document.xml' section of a MSWord .dotm file.
 """
-__author__ = "???"
+__author__ = "knmarvel"
+#   time spent on this project: 2 hrs
+
+
+def examine_files(dir, char):
+    files = {}
+    total_files_searched = 0
+    for filename in os.listdir(dir):
+        filepath = dir + "/" + filename
+        if zipfile.is_zipfile(filepath):
+            with zipfile.ZipFile(filepath, 'r') as zip_ref:
+                zip_ref.extractall("./extracted")
+            with open('./extracted/word/document.xml', "r") as doc:
+                doc = doc.read()
+            if "office/word" in doc:
+                total_files_searched += 1
+                if char in doc:
+                    files[filename] = doc[doc.index(char) - 40: doc.index(char) + 40]
+    f_srched = "# of files searched: " + str(total_files_searched)
+    f_fnd = "# of files found:" + str(len(files))
+    f = "\n".join([": ".join([x, files[x]]) for x in files])
+    return (f_srched + "\n" + f_fnd + "\n" + f)
 
 
 def main():
-    raise NotImplementedError("Your awesome code begins here!")
+    parser.add_argument('--dir', help='directory to find data',
```

Comment:
maybe " help='the directory to be searched' " might be a little clearer?

Reply:
Actually changed to "directory of files to search"